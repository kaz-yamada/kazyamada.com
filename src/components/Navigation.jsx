import React, { Component } from "react";

import { Link } from "gatsby";

export default class Navigation extends Component {
  constructor(props) {
    super(props);

    this.state = {
      showMenu: false,
      menuClass: "",
    };
  }

  toggleMenu = () => {
    this.setState((prevState) => {
      const menuClass = prevState.showMenu ? "" : "show-menu";

      return { showMenu: !prevState.showMenu, menuClass };
    });
  };

  onMenuButtonClick = (e) => {
    e.preventDefault();
    this.toggleMenu();
  };

  hideMenu = () => {
    this.setState({ showMenu: false, menuClass: "" });
  };

  render() {
    const { config } = this.props;
    const { menuClass } = this.state;
    const { menuLinks } = config;

    return (
      <nav className={`main-navigation ${menuClass}`}>
        <button
          id="menu-button"
          className="menu-button"
          type="button"
          onClick={this.onMenuButtonClick}
        >
          <div className="menu-icon" />
        </button>
        <div className="navbar">
          <div className="nav-title">
            <Link className="menu-link" activeClassName="active" to="/#">
              {config.siteTitleShort}
            </Link>
          </div>
          <div className="menu-item">
            {menuLinks.map((item) => {
              return (
                <Link
                  className="menu-link"
                  activeClassName="active"
                  key={item.name}
                  to={item.link}
                  onClick={this.hideMenu}
                >
                  {item.name}
                </Link>
              );
            })}
          </div>
        </div>
      </nav>
    );
  }
}
