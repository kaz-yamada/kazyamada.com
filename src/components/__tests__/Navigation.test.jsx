import React from "react";

import renderer from "react-test-renderer";

import Navigation from "../Navigation";
import config from "../../../data/SiteConfig";

describe("Navigation bar", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Navigation config={config} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
