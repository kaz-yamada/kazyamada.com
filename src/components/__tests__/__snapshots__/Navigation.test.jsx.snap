// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar renders correctly 1`] = `
<nav
  className="main-navigation "
>
  <button
    className="menu-button"
    id="menu-button"
    onClick={[Function]}
    type="button"
  >
    <div
      className="menu-icon"
    />
  </button>
  <div
    className="navbar"
  >
    <div
      className="nav-title"
    >
      <a
        className="menu-link"
        href="/#"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Kaz Yamada
      </a>
    </div>
    <div
      className="menu-item"
    >
      <a
        className="menu-link"
        href="/#about-me"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        About
      </a>
      <a
        className="menu-link"
        href="/#projects"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Projects
      </a>
      <a
        className="menu-link"
        href="/blog/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Blog
      </a>
    </div>
  </div>
</nav>
`;
